posts/Yoann/2019-04-05-hibernate-search-6-0-0-Alpha4.adoc

```diff
@@ -0,0 +1,149 @@
+:awestruct-layout: blog-post
+---
+
+We just published Hibernate Search 6.0.0.Alpha4, the fourth release for the still-in-development 6.0 branch.
+This release mainly adds an "exists" predicate,
+improves the DSL by allowing to bypass DSL and projection converters or to override analyzers,
+restores the Elasticsearch AWS integration,
+and upgrades the Lucene backend to Lucene 8.
+
++++<!-- more -->+++
+
+== Getting started with Hibernate Search 6
+
+If you want to dive right into the new, shiny Hibernate Search 6,
+a good starting point is the
+https://docs.jboss.org/hibernate/search/6.0/reference/en-US/html_single/#getting-started[getting started guide]
+included in the reference documentation.
+
+[NOTE]
+====
+Hibernate Search 6 is still in development and its APIs differ significantly from Search 5.
+
+For more information about the current status of this branch, see
+http://hibernate.org/search/releases/6.0/#whats-new[the page dedicated to Search 6 on hibernate.org].
+
+For more information about migration and what we intend to do to help you, see
+http://hibernate.org/search/documentation/migrate/6.0/[the migration guide].
+====
+
+== What's new
+
+=== New "exists" predicate
+
+We added support for the "exists" predicate in the Search DSL,
+allowing to filter documents according to the presence or absence of a non-`null` value for a given field:
+
+[source, JAVA, indent=0]
+----
+SearchQuery<MyEntity> query = searchSession.search(MyEntity.class).asEntity()
+    .predicate(f -> f.exists().onField("someField"))
+    .toQuery();
+----
+
+=== The Elasticsearch AWS integration is back
+
+We restored the AWS integration as part of https://hibernate.atlassian.net/browse/HSEARCH-3051[HSEARCH-3051].
+See https://docs.jboss.org/hibernate/search/6.0/reference/en-US/html_single/#backend-elasticsearch-configuration-aws[the dedicated section in the documentation]
+to configure your backend to talk with an AWS-managed Elasticsearch service.
+
+=== Converters can be bypassed in predicates
+
+DSL and projection converters automatically convert values from the type expected by the user to the underlying type exposed by backends.
+For example `java.util.Date` properties have a converter between that type and `Instant`.
+
+Just like the `.ignoreFieldBridge` option in Search 5,
+https://hibernate.atlassian.net/browse/HSEARCH-3510[there is now an option] to bypass these converters in Search 6:
+
+[source, JAVA, indent=0]
+----
+// When the user input is a java.util.Date
+SearchQuery<MyEntity> query = searchSession.search(MyEntity.class).asEntity()
+    .predicate(f -> f.match().onField("myJavaUtilDate").matching(javaUtilDateUserInput))
+    .toQuery();
+// When the user input is a java.time.Instant
+SearchQuery<MyEntity> query = searchSession.search(MyEntity.class).asEntity()
+    .predicate(f -> f.match().onField("myJavaUtilDate").matching(instantUserInput, DslConverter.DISABLED))
+    .toQuery();
+----
+
```

Comment:
I would add:
the user can choose to receive in a projection the value in the original or in the underlying backend format.